src/provider.rs

```diff
@@ -448,6 +468,24 @@ impl Handler {
             version: env!("CARGO_PKG_VERSION").to_string(),
         }
     }
+    async fn id(self, _: IdRequest) -> IdResponse {
+        IdResponse {
+            peer_id: Box::new(self.peer_id),
```

Comment:
what's the reason for this boxing?

Reply:
Otherwise you get an enum variant size mismatch warning, because these things are large.

Reply:
oh, the whole `IdResponse` becomes too large.  May want to explain that in the rpc_protocol.rs file with a comment.  Does it make sense to put everything in one "inner" box instead?